CONTROLLERS.md

```diff
@@ -35,5 +35,7 @@ default values, and whether a property supports the NiFi Expression Language.
 
 | Name | Default Value | Allowable Values | Expression Language Supported? | Description |
 | - | - | - | - | - |
-| **Access Key** | | | Yes | Specifies the AWS Access Key |
-| **Secret Key** | | | Yes | Specifies the AWS Secret Key |
+|**Use Default Credentials**|false||No|If true, uses the Default Credential chain, including EC2 instance profiles or roles, environment variables, default user credentials, etc.|
+|Access Key|||Yes|Specifies the AWS Access Key|
```

Comment:
Access key being required seems to be a bit strange to me when I can use default credentials.

I have a feeling that we need custom validation here: either both keys should be provided or default cred should be used.

Reply:
I actually changed it to not be required anymore (removed the * in the docs). The validation starts with the Use Default Credentials, if it is set it will go through the AWS chain and try to get the credentials from there. If it's not found then it checks if the Access Key and Secret Key are set and uses them as credentials if both are set. If those are not set either it checks if the Credentials File is set and tries to read the access key and secret key properties from that file.

Reply:
Okay, looks good, thanks!